cleverhans/utils.py

```diff
@@ -220,6 +220,97 @@ def grid_visual(data):
     return figure
 
 
+def get_logits_over_interval(sess, model, x, adv_x, x_data,
+                             min_epsilon=-10, max_epsilon=10,
+                             num_points=21):
+    """Get logits when the input is perturbed in an interval in adv direction.
+
+    Args:
+        sess: Tf session
+        model: Model for which we wish to get logits
+        x: Data tensor
+        adv_x: Tensor for adversarial perturbation of x
+        x_data: Numpy array corresponding to the data
+            slice used to generate plot
+        min_epsilon: Minimum value of epsilon over the interval
+        max_epsilon: Maximum value of epsilon over the interval
+        num_points: Number of points used to interpolate
+
+    Returns:
+        Numpy array containing logits
+
+    Raises:
+        ValueError if min_epsilon is larger than max_epsilon
+    """
+    import tensorflow as tf
+    if min_epsilon > max_epsilon:
+        raise ValueError('Minimum epsilon is less than maximum epsilon')
+
+    epsilon = min_epsilon
+    eta = adv_x - x
+    eta = tf.nn.l2_normalize(eta, dim=0)
+    for i, epsilon in enumerate(np.linspace(min_epsilon,
```

Comment:
if x is a single example, it could be possible to do this without a for loop, and with only 1 sess.run call rather than 21. You would make one batch containing all the different sizes of perturbation and run them in a single sess.run call.

Reply:
I updated it do it over one sess call. However the lep now looks like this: 

![lep_adv3](https://user-images.githubusercontent.com/8907160/30940210-1a949896-a395-11e7-878c-238fc187baa3.png)

As opposed to this:
![lep_adv](https://user-images.githubusercontent.com/8907160/30940175-f9d5edb2-a394-11e7-9e69-935810bd15cd.png)

I am wondering if there is an issue in the way I do it.